File: js/task3a/S3_OBJ_01.js
function get(obj, path, fallback) {
  const keys = path.split('.');

  let current = obj;
  for (let key of keys) {
      if (current === null || current === undefined || !(key in current)) {
          return fallback;
      }
      current = current[key];
  }

  return current;
}

const data = { a: { b: { c: 42 } }, x: null };

console.log(get(data, 'a.b.c', 0));    
console.log(get(data, 'a.b.d', 0));     
console.log(get(data, 'x.y', 0));       
console.log(get(data, '', 0));         

File: python/task3a/S3_OBJ_01.py
def get(obj, path, fallback):
  if path == "":
      return fallback

  keys = path.split(".")
  current = obj

  for key in keys:
      if current is None or key not in current:
          return fallback
      current = current[key]

  return current


data = { "a": { "b": { "c": 42 } }, "x": None }

print(get(data, "a.b.c", 0))
print(get(data, "a.b.d", 0))
print(get(data, "x.y", 0))
print(get(data, "", 0))
